tests/util.bats

```diff
@@ -0,0 +1,383 @@
+#!/usr/bin/env bats
```

Comment:
Added some tests for the new functions I added, so it's easier to test and maintain their functionality through time.
The tests can be run with [bats](https://github.com/bats-core/bats-core) by calling `bats tests/` from the root of the repository. Ideally, you'd want to run the tests with a CI by calling `bats --tap tests/` (which shows the TAP-compliant output for tests)